pom.xml

```diff
@@ -564,6 +564,16 @@
                                         <wait>
                                             <time>40000</time>
                                             <log>Service Broker manager has started</log>
+                                            <exec>
+                                                <!--
+                                                Prepare a new "pdb" database for tests with READ_COMMITTED_SNAPSHOT enabled
+                                                 (required for SQL server to have the same behavior as other DBs in READ_COMMITTED
+                                                 isolation level) and ALLOW_SNAPSHOT_ISOLATION enabled to allow using the
+                                                 TRANSACTION_SNAPSHOT level (equivalent to SERIALIZABLE in PostgreSQL and Oracle).
+                                                A new database needs to be used because the flags can't be enabled on "master" database.
+                                                -->
+                                                <postStart>/opt/mssql-tools/bin/sqlcmd -l 45 -U sa -P AaBb12.# -Q CREATE\ DATABASE\ pdb;\ ALTER\ DATABASE\ pdb\ SET\ READ_COMMITTED_SNAPSHOT\ ON;\ ALTER\ DATABASE\ pdb\ SET\ ALLOW_SNAPSHOT_ISOLATION\ ON</postStart>
```

Comment:
What's the advantage of doing this here vs in some initializer of the tests in Java?
Here feels more like magic that developers will tend to forget about easily.

Reply:
Let's call it "transparent" to the user/developer.
The docker initialization performed here reflects how a production database and PDB should work together -- in this case of SQL Server, PDB is expected to work with a non-"master" database (which is recommended anyway) that needs to have at least READ_COMMITTED_SNAPSHOT enabled in order to work like the other DB engines.
This same requirement is mentioned in the "quirks" page in the PDB wiki.

This is transparent in the sense that, while these settings in the pom are not readily accessible from Java code and are easy to forget about, in reality they only need to be in sync with the contents of the "connections.properties" file (usernames, passwords, database, etc.) to be used in the tests.
In this particular case, creating the "pdb" database would require some code to run before all the tests, establishing an initial connection to the "master" database just to create the new one (note that we're using the vanilla docker containers and we're not using doker-java :smile: ).